core/src/main/java/org/mule/runtime/core/api/routing/SelectiveRouter.java

```diff
@@ -18,11 +18,11 @@
  */
 public interface SelectiveRouter extends Router {
 
-  void addRoute(String expression, MessageProcessorChain processor);
+  void addRoute(String expression, Processor processor);
```

Comment:
Why? Isn't it best that routers work with child chains rather than processors?  If they use child chains then we know processing strategy and notifications will be fired/used correctly, whereas with Processor the router than needs to create these chains internally.